subsys/bluetooth/host/hci_core.c

```diff
@@ -7989,12 +7990,21 @@ void bt_le_adv_resume(void)
 	err = set_le_adv_enable(adv, true);
 	if (err) {
 		bt_conn_set_state(conn, BT_CONN_DISCONNECTED);
+
```

Comment:
Unrelated to this patch, but I think it'd be good to have a BT_WARN or BT_ERR here logging the exact `err` value. There are a couple of other places where this function returns which also look more like BT_WARN or BT_ERR to me rather than BT_DBG that you've used. Anyway, as I said unrelated and therefore not blocking's this PR.

Reply:
I wouldn't want to use BT_WARN in this case because this is the expected behaviour.
For example above the host is not resuming the connectable advertiser because the host has no more connection objects. 
And the controller might deny starting the connectable advertiser if the controller has no more connection objects.

This is the case for us now, because the controller has one slave connection and one central connection available, but the host has 2 connection objects (because it does not distinguish type). This actually gives us a warning from the HCI command.

I can add BT_INFO, then it is more visible atleast. And change the one above to BT_INFO, does that make sense?

Reply:
Not BT_INFO. We should be super conservative about it since it's always enabled when logging is enabled. BT_DBG perhaps. That said, if there's a mismatch of number of connections supported between host and controller, doesn't that imply that you have a suboptimal build configuration on either side? If that's the case then it still feels like BT_WARN would be warranted.

Reply:
> That said, if there's a mismatch of number of connections supported between host and controller, doesn't that imply that you have a suboptimal build configuration on either side?

Not for our out-of-tree maintained controller, which has a BT_CONN_SLAVE_MAX_COUNT and a BT_CONN_MASTER_MAX_COUNT, and BT_CONN_MAX_COUNT needs to be a sum of those.
This is why we ended up triggering this scenario.

I'll add a BT_DBG line for it.

Reply:
Added BT_DBG for when the controller cannot start the advertiser.